ivy/stateful/module.py

```diff
@@ -525,16 +529,33 @@ def __getattribute__(self, name):
     def __setattr__(self, name, value):
         if name in ["v", "buffers"]:
             name = "_" + name
-        if isinstance(value, Module):
-            ret = super().__setattr__(name, value)
+
+        from ivy.functional.ivy.gradients import _is_variable
+
+        if (
+            name != "_v" and _is_variable(value.ivy_array)
+            if hasattr(value, "_ivy_array")
+            else (ivy.is_array(value) and _is_variable(value))
+        ):
+            if hasattr(self, "_v"):
+                self._v[name] = value
+            if (
+                hasattr(self, "_build_mode")
+                and self.build_mode == "on_init"
+                and self.built
+            ):
+                self._rebuild()
+
+        elif name != "_module_dict" and isinstance(value, Module):
```

Comment:
Also a bit regarding this, I don't see any harm in adding this check but if we do end up with a case where the `name` is `_module_dict` and the `value` is an `ivy.Module`, then I think there's a bug somewhere else to be fixed because ideally `_module_dict` should always be a `Container` 😅